src/components/QuestionView.jsx

```diff
@@ -1,72 +1,132 @@
 import React, { useState } from "react";
 
-const QuestionView = ({ question, index, setIndex, limit }) => {
+import ScorePage from "./ScorePage";
+
+const QuestionView = ({ question, correctAns, index, setIndex, limit }) => {
+  //const givenAns = localStorage.getItem(question.id)
   const defaultFormData = {
     answer_a_correct: "false",
     answer_b_correct: "false",
     answer_c_correct: "false",
     answer_d_correct: "false",
-    answer_e_correct: "false",
+    answer_e_correct: "false", 
     answer_f_correct: "false",
   };
   const [score, setScore] = useState(0);
   const [skip, setSkip] = useState(0);
   const [formData, setFormData] = useState(defaultFormData);
+  const [givenAns, setGivenAns] = useState({}); 
+
+ 
+
   const handleChange = (e) => {
     setFormData({ ...formData, [e.target.name]: e.target.checked.toString() });
   };
 
   const next = () => {
-    checkAnswer(formData, question.correct_answers);
+    // checkAnswer(formData, question.correct_answers);
+    storeAnswer(formData);
     setFormData(defaultFormData);
     setIndex(index + 1);
   };
 
-  const checkAnswer = (options, answer) => {
-    JSON.stringify(options) === JSON.stringify(answer)
-      ? setScore(score + 1)
-      : setScore(score - 0.5);
+  const prev = () => {
+    // checkAnswer(formData, question.correct_answers);
+    storeAnswer(formData);
+    setFormData(defaultFormData);
+    setIndex(index - 1);
+  };
+
+  const calculateScore = () => {
+
+    // console.log("score")
+    // return Object.keys(correctAns).reduce((acc, questionId) => {
+    //     JSON.stringify(correctAns[questionId]) === JSON.stringify(givenAns[questionId]) && acc++
+    //     return acc
+    // }, 0)  
+    
+    return Object.keys(givenAns).reduce((acc, questionId) => {
+      let flag=true
+      console.log(givenAns)
+      Object.entries(givenAns[questionId]).forEach(([key,value])=>{
+        //console.log(key)
+        //console.log(correctAns[questionId][key])
+         
```

Comment:
Remove unnecessary line breaks. Format the code before pushing.
Add a script that will automatically do it for you.